gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/schema/ZoweJwtScheme.java

```diff
@@ -9,23 +9,92 @@
  */
 package org.zowe.apiml.gateway.security.service.schema;
 
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.zuul.context.RequestContext;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import org.apache.http.HttpRequest;
 import org.springframework.stereotype.Component;
 import org.zowe.apiml.auth.Authentication;
 import org.zowe.apiml.auth.AuthenticationScheme;
 import org.zowe.apiml.gateway.security.service.schema.source.AuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.AuthSourceService;
+import org.zowe.apiml.gateway.security.service.schema.source.UserNotMappedException;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.apiml.security.common.config.AuthConfigurationProperties;
+import org.zowe.apiml.security.common.error.AuthenticationTokenException;
+import org.zowe.apiml.util.Cookies;
+
+import java.util.Date;
+import java.util.Optional;
 
 
 @Component
+@AllArgsConstructor
 public class ZoweJwtScheme implements AbstractAuthenticationScheme {
 
+    private AuthSourceService authSourceService;
+    private AuthConfigurationProperties configurationProperties;
+    private MessageService messageService;
+
     @Override
     public AuthenticationScheme getScheme() {
         return AuthenticationScheme.ZOWE_JWT;
     }
 
+    @Override
+    public Optional<AuthSource> getAuthSource() {
+        return authSourceService.getAuthSourceFromRequest();
+    }
+
     @Override
     public AuthenticationCommand createCommand(Authentication authentication, AuthSource authSource) {
-        return AuthenticationCommand.EMPTY;
+        final AuthSource.Parsed parsedAuthSource = authSourceService.parse(authSource);
+        String error = null;
+        String jwt = null;
+        if (authSource == null || authSource.getRawSource() == null) {
+            error = this.messageService.createMessage("org.zowe.apiml.gateway.security.schema.missingAuthentication").mapToLogMessage();
+            return new ZoweJwtAuthCommand(null, null, error);
+        }
+        final Date expiration = parsedAuthSource == null ? null : parsedAuthSource.getExpiration();
+        final Long expirationTime = expiration == null ? null : expiration.getTime();
+        try {
+            jwt = authSourceService.getJWT(authSource);
+        } catch (UserNotMappedException | AuthenticationTokenException e) {
+            error = this.messageService.createMessage(e.getMessage()).mapToLogMessage();
```

Comment:
Will it be able to create the message anyway, even if you're not passing a message key defined in the messages.yml?

Reply:
for now, error handling is done only for known issues and will be consolidated once we have all auth schemes updated